api/src/convert/tx_in.rs

```diff
@@ -40,7 +44,53 @@ impl TryFrom<&external::TxIn> for tx::TxIn {
             proofs.push(tx_proof);
         }
 
-        let tx_in = tx::TxIn { ring, proofs };
+        let input_rules = source
+            .input_rules
+            .as_ref()
+            .map(InputRules::try_from)
+            .transpose()?;
```

Comment:
Why do we need the transpose here?

Reply:
transpose is a thing that turns `Option<Result<T>>` into `Result<Option<T>>`, if I do `.transpose()?` it has the effect of pulling an error case out of `Option<Result<T>>` and returning it in a fairly succinct way.

Reply:
Oh that's a neat trick. Thank you. I should use that.